### PR TITLE
Fix FreeBSD build

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,7 +98,11 @@ else (WIN32)
     enable_language(ASM)
 
     # Ensure that awk is present
-    find_program(AWK awk)
+    if (CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
```
Comment (janvorli) [inline CMakeLists.txt]: This change is not needed, there is awk on FreeBSD too. If you want to enable gawk as an option for people that don't want to have awk installed, then please add a second `find_program` just without `if`. The `find_program` looks for the target program only when the `AWK` is not set yet or it is set to `AWK-NOTFOUND`.

File: src/gc/unix/gcenv.unix.cpp [hunk 2]
```diff
@@ -7,6 +7,9 @@
 #include <cassert>
 #include <memory>
 #include <pthread.h>
+#ifdef __FreeBSD__
```
Comment (janvorli) [inline src/gc/unix/gcenv.unix.cpp]: I can see that we differ in the way to get the thread id here and in the https://github.com/dotnet/coreclr/blob/master/src/pal/src/include/pal/thread.hpp#L838-L844. Could you please unify it? If both ways return the same values, then I'd use the pthread_getthreadid_np at both places. If not, then please let's use the `thr_self` as it is verified to provide the same identifier as LLDB, which we need in the LLDB plugin.

**Review by janvorli — COMMENTED** [review]
Thank you for fixing the build, I have a couple of comments.

File: CMakeLists.txt (hunk 1 above)
Comment (devnexen) [reply CMakeLists.txt]: In fact it was to avoid the risk of different output (in future) between bsd awk and gnu awk but I can revert it no problems.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Comment by stephentoub** [comment]
cc: @wfurt

**Comment by stephentoub** [comment]
@wfurt, should this be merged? Closed?

**Comment by wfurt** [comment]
Current master should build on FreeBSD. However pthread_getthreadid_np() seems cleaner to me. I'm not sure if there is any practical impact. We also had issues with threading on 12.0. I don't know if this can have any impact.  Do you have any ideas @devnexen  ???

**Comment by devnexen** [comment]
API-wise I would also prefer pthread_getthreadid_np but in this case the outcome should not change much regarding the use case here.

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in dotnet/coreclr#27549 this repository will be moving to dotnet/runtime on November 13. If you would like to continue working on this PR after this date, the easiest way to move the change to dotnet/runtime is:
1.	In your coreclr repository clone, create patch by running `git format-patch origin`
2.	In your runtime repository clone, apply the patch by running `git apply --directory src/coreclr <path to the patch created in step 1>`

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in #27549 the dotnet/runtime repository will be used going forward for changes to this code base. Closing this PR as no more changes will be accepted into master for this repository. If you’d like to continue working on this change please move it to dotnet/runtime.

**Pull request closed without merging**